src/include/distributed/metadata_utility.h

```diff
@@ -201,6 +201,37 @@ typedef enum SizeQueryType
 	TABLE_SIZE /* pg_table_size() */
 } SizeQueryType;
 
+
+typedef enum
+{
+	COLOCATE_WITH_TABLE_LIKE_OPT,
+	COLOCATE_WITH_COLOCATION_ID
+} ColocationParamType;
+
+/*
+ * Param used to specify the colocation target of a distributed table. It can
+ * be either a table name or a colocation id.
+ *
+ * When colocationParamType is COLOCATE_WITH_COLOCATION_ID, colocationId is
+ * expected to be a valid colocation id. When colocationParamType is set to
+ * COLOCATE_WITH_TABLE_LIKE_OPT, colocateWithTableName is expected to
+ * be a valid table name, "default" or "none".
+ *
+ * Among the functions used to create a Citus table, right now only
+ * CreateSingleShardTable() accepts a ColocationParam.
+ */
+typedef struct
+{
+	union
+	{
+		char *colocateWithTableName;
+		uint32 colocationId;
+	};
```

Comment:
if it simplifies the syntax, we can afford to have both fields in top-level struct